openfl/component/director/director.py

```diff
@@ -114,33 +120,40 @@ def get_registered_shards(self) -> list:
         """Get registered shard infos."""
         return [shard_status['shard_info'] for shard_status in self._shard_registry.values()]
 
-    async def stream_metrics(self, experiment_name: str, caller: str):
-        """Stream metrics from the aggregator."""
+    def get_next_metric(self, experiment_name: str, caller: str):
+        """
+        Stream metrics from the aggregator.
+
+        This method takes next metric dictionary from the aggregator's queue
+        and returns it to the caller.
+
+        Inputs:
+            experiment_name - string id for experiment
+            caller - string id for experiment owner
+
+        Returns:
+            metric_dict - {'metric_origin','task_name','metric_name','metric_value','round'}
+                if the queue is not empty
+            None - f queue is empty but the experiment is still running
+
+        Raises:
+            StopIteration - if the experiment is finished and there is no more metrics to report
+        """
         aggregator = self.experiment_stash[caller][experiment_name].aggregator
+        while True:
+            if not aggregator.metric_queue.empty():
+                yield aggregator.metric_queue.get()
+                continue
+
+            if aggregator.all_quit_jobs_sent():
```

Comment:
We should also check that `aggregator.metric_queue.empty()` here, because after last check it could be a switch to the thread in which the aggregator was executed and new data got into the queue.

Reply:
done!